python/mxnet/optimizer/optimizer.py

```diff
@@ -581,7 +581,9 @@ def create_state(self, index, weight):
         momentum = None
         if self.momentum != 0.0:
             stype = weight.stype if self.lazy_update else 'default'
-            momentum = zeros(weight.shape, weight.context, dtype=weight.dtype, stype=stype)
+            momentum = zeros(weight.shape, weight.context, dtype=weight.dtype, stype=stype,
+                             name=weight.name.replace("in_arg", "optimizer_state:momentum"))
```

Comment:
This seems a bit adhoc

Reply:
Please let us know if there is any better way of annotating those optimizer states.